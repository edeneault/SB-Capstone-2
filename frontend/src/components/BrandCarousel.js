import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { Image } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import Loader from "./Loader";
import Message from "./Message";
import { listBrandProducts } from "../actions/productActions";
import "../css/ProductCarousel.css";
import Carousel from "react-multi-carousel";
import shuffle from "../utils/shuffle";

const responsive = {
  desktop: {
    breakpoint: { max: 3000, min: 1024 },
    items: 3,
    paritialVisibilityGutter: 60,
  },
  tablet: {
    breakpoint: { max: 1024, min: 464 },
    items: 2,
    paritialVisibilityGutter: 50,
  },
  mobile: {
    breakpoint: { max: 464, min: 0 },
    items: 1,
    paritialVisibilityGutter: 30,
  },
};

const BrandCarousel = ({ brand }) => {
  const dispatch = useDispatch();

  const productBrand = useSelector((state) => state.productBrand);
  const { loading, error, products } = productBrand;
  shuffle(products);

  useEffect(() => {
    dispatch(listBrandProducts(brand));
  }, [dispatch, brand]);

  return loading ? (
    <Loader />
  ) : error ? (
    <Message variant='danger'>{error}</Message>
  ) : (
    <>
      <h2 className='carousel-title m-0 p-3 text-dark fs-1'>
        ~ Brand {brand} ~
      </h2>
      <Carousel
        itemClass='image-item'
        responsive={responsive}
        swipeable
        renderArrowsWhenDisabled
        // infinite
        centerMode
        className='fade-in'
      >
        {products.slice(0, 12).map((product) => {
          return (
            <div key={product.name} className=''>
              <Image
                fluid
                className='img-carousel'
                draggable={false}
                src={product.image}
              />
              <Link to={`/products/${product._id}`} className='text-dark'>
                <p className='p-0 m-0  fs-6 '>{product.name}</p>
                <p className='p-0 m-0  fs-6'>{product.price}</p>
              </Link>
            </div>
          );
        })}
      </Carousel>
    </>
  );
};

export default BrandCarousel;
