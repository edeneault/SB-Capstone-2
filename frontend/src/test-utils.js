import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
