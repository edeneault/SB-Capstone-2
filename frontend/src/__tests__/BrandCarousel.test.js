import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";

import BrandCarousel from "../components/BrandCarousel.js";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  test("renders BrandCarousel without crashing", async () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <BrandCarousel brand='lavazza' />
      </Provider>,
    );

    expect(getByText("~ Brand lavazza ~")).toBeDefined();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <BrandCarousel brand='lavazza' />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
