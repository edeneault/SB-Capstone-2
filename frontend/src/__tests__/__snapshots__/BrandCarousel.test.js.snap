// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <h2
    class="carousel-title m-0 p-3 text-dark fs-1"
  >
    ~ Brand lavazza ~
  </h2>
  <div
    class="react-multi-carousel-list  fade-in"
  >
    <ul
      class="react-multi-carousel-track "
      style="transition: none; overflow: unset; transform: translate3d(200px,0,0);"
    />
  </div>
</DocumentFragment>
`;
